test/Microsoft.AspNetCore.Authentication.Test/CollectingEventListener.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Authentication.Test
```

Comment:
How important do we think testing of these events really is?

Reply:
That's part of the discussions here. I don't know if we really care about testing the events but I wanted to sketch out what it would look like. It's not much different from testing logging and I don't think we do that very much either,.

Reply:
yeah I'm not afraid to take a stand against testing these. I think the pattern that you showed here is reasonable, but I think overall it's kinda low value. Doing a good job testing these usually involve injecting error into places and since we have a really simple error handling story we usually don't do that today so we have to invent it.

Reply:
How do we know that we didn't regress things?